<?xml version="1.0" encoding="utf-8"?>

<launch>

    <arg name="robot_name" default="ugv1" />
    <arg name="simulator" default="/vrep" />

    <arg name="respawn_value" default="false" /> <!-- boolean: true, false -->

    <arg name="param_file" default="$(find scan_space_time_filter)/launch/scan_filter2.yaml" />

    <node name="scan_filter2_$(arg robot_name)" pkg="scan_space_time_filter" type="scan_filter_node2" respawn="$(arg respawn_value)" output="screen">

        <rosparam command="load" file="$(arg param_file)" />

        <param name="simulator" value="$(arg simulator)" />

        <param name="robot_name" value="$(arg robot_name)" />

        <remap from="dynamic_point_cloud1" to="$(arg simulator)/$(arg robot_name)/depth_pcl" /> <!-- input 1 -->
        <remap from="dynamic_point_cloud2" to="$(arg simulator)/$(arg robot_name)/depth_pcl2" /> <!-- input 2 -->        

        <remap from="filtered_pointcloud" to="$(arg simulator)/$(arg robot_name)/filtered_pointcloud" /> <!-- output -->

    </node>


</launch>
